apps/rush-lib/src/cli/actions/GenerateAction.ts

```diff
@@ -105,20 +108,27 @@ export default class GenerateAction extends BaseRushAction {
       // Do an incremental install
       installManager.installCommonModules(InstallType.Normal);
 
-      console.log(os.EOL + colors.bold('(Skipping "npm shrinkwrap")') + os.EOL);
+      if (this.rushConfiguration.packageManager === 'npm') {
+        console.log(os.EOL + colors.bold('(Skipping "npm shrinkwrap")') + os.EOL);
+      } else if (this.rushConfiguration.packageManager === 'pnpm') {
+        console.log(os.EOL + colors.bold('(Deleting "shrinkwrap.yaml")') + os.EOL);
+        fsx.unlinkSync(tempShrinkwrapFilename);
+      }
     } else {
       // Do a clean install
```

Comment:
> // Do a clean install [](start = 5, length = 22)

We should provide a "--clean" option for PNPM, in case the store collects deadwood or gets corrupted.

Otherwise, the "isLazy" branch should be our default behavior for PNPM.